File: 1399A.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define pb push_back
const ll MAX = 100;


void solve(){
    int n;
    cin >> n;
    vector<int> input(n);
    for(int i = 0; i<n; i++){
        cin >> input[i];
    }
    sort(input.begin(), input.end());
    for(int i = 0; i<n-1; i++){
        if(abs(input[i]-input[i+1]) > 1){
            cout << "NO" << endl;
            return;
        }
    }
    cout << "YES" << endl;
}

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	bool testcases = true;
	if(testcases){
		int t;
		cin >> t;
		while(t--){
			solve();
		}
	}
	else{
		solve();
	}
	return 0;	
}
 

File: 1399A.py
import math
def BubbleSort(tab):
    n = len(tab)
    for i in range(n - 1):
        for j in range(0, n - i - 1):
            if tab[j] > tab[j + 1]:
                tab[j], tab[j + 1] = tab[j + 1], tab[j]
t = int(input())
for i in range(1, t + 1):
    l = int(input())
    a = list(map(int, input().split()))
    j = 0
    b = []
    BubbleSort(a)
    for j in range(0, len(a)):
        try:
            if abs(a[j] - a[j + 1]) <= 1:
                if a[j] < a[j + 1]:
                    b.append(a[j])
                    j += 1
                elif a[j] > a[j + 1]:
                    b.append(a[j + 1])
                    j += 1
                else:
                    b.append(a[j + 1])
                    j += 1
        except:
            pass

    if len(a) - len(b) == 1:
        print("YES")
    else:
        print("NO")